modules/n1ql/pages/n1ql-language-reference/explainfunction.adoc

```diff
@@ -0,0 +1,105 @@
+= EXPLAIN FUNCTION
+:description: EXPLAIN FUNCTION provides information about the execution plan for the {sqlpp} subqueries or embedded statements defined for a specified user-defined function. 
+:page-topic-type: reference
+:imagesdir: ../../assets/images
+
+// TEMP
+include::partial$n1ql-language-reference/collapsible-style.adoc[]
+
+[abstract]
+{description}
+
+You can supply the name of an inline or external user-defined function.
+
+* For an inline function, EXPLAIN FUNCTION returns the query plans for all of the subqueries present in the function body.
++
+For more information about inline functions, see xref:n1ql-language-reference:userfun.adoc[].
+
+* For an external function, EXPLAIN FUNCTION returns the query plans for all embedded {sqlpp} queries inside the referenced JavaScript body, or the line number on which a N1QL() call appears.
++
+For more information about user-defined functions with JavaScript, see xref:guides:javascript-udfs.adoc[].
+
+The following constraints apply:
+
+* If a user-defined function itself contains other, nested user-defined function executions, EXPLAIN FUNCTION generates the query plan for the specified function only, and not for its nested {sqlpp} queries.
+
+* If an external function defines an alias for an N1QL() call, EXPLAIN FUNCTION cannot return its line number.
+
+== Prerequisites
+
+To execute EXPLAIN FUNCTION on an inline user-defined function, you must have the necessary execute function permission on that function. 
+
+* For a global internal function, you must have the Execute Global Functions role. 
+
+* For a scoped internal function, you must have the Execute Scope Functions role, with permissions on the specified bucket and scope.
+
+To execute EXPLAIN FUNCTION on an external user-defined JavaScript function, you must have the necessary execute external function permission on that function. 
+
+* For a global JavaScript function, you must have the Execute Global External Functions role. 
+
+* For a scoped JavaScript function, you must have the Execute External Scope Functions role, with permissions on the specified bucket and scope.
+
+You must also have the necessary privileges required for the {sqlpp} statements inside the function.
+
+For more information about user roles, see
+xref:learn:security/authorization-overview.adoc[Authorization].
+
+== Syntax
+
+[source,ebnf]
+----
+include::partial$grammar/utility.ebnf[tag=explainfunction]
+----
+
+image::n1ql-language-reference/explainfunction.png["Syntax diagram: refer to source code listing", align=left]
+
+The statement consists of the `EXPLAIN FUNCTION` keywords followed by the name of the user-defined function that has the execution plan you want to review.
+
+== Examples
+
+=== Inline Function Example
+
+In this example, you create an inline function named `func1` and then request the plan for its subquery.
+
+====
```

Comment:
We usually use titles for the examples in the references rather than individual headings. Also suggest putting the "in this example" intro paragraph into the actual example.

```suggestion
.Inline Function Example

====
In this example, you create an inline function named `func1` and then request the plan for the embedded statement.

```